PerformanceTest_Static_Pressure_PlantarDorsal_ShinCalf_v2.py

```diff
@@ -0,0 +1,413 @@
+    ax4.set(xticklabels=[])
+    ax4.set_title('Calf Pressure')  
+    
+    plt.suptitle(inputFD.config)
+    plt.tight_layout() 
+    
+    saveFolder= FilePath + '2DPlots'
+    
+    if os.path.exists(saveFolder) == False:
+        os.mkdir(saveFolder)
+        
+    plt.savefig(saveFolder + '/' + inputFD.subject + inputFD.config + '.png')
+    return fig
+
+
+
+
+config = []
+subject = []
+Movement = []
+badFileList = []
+
+plantarContact = []
+plantarPeakPressure = []
+plantarAvgPressure = []
+plantarSDPressure = []
+plantarTotalPressure = []
+
+heelArea = [] 
+heelAreaUP = []
+
+parentList = entries_FootDorsum 
+
+fd_outcomes = []
+ 
+
+# check if front back has more data
+if len(entries_FrontBack) > len(entries_FootDorsum):    
+    for entry in entries_FrontBack: 
+        subject = entry.split(sep="_")[0] 
+        config = entry.split(sep="_")[1]
+        trial = entry.split(sep="_")[3].split(sep='.')[0]   
+        counter = 0
+        for ii in range(len(entries_FootDorsum)):
+            sub2 = entries_FootDorsum[ii].split(sep="_")[0]
+            config2 = entries_FootDorsum[ii].split(sep="_")[1]
+            trial2 = entries_FootDorsum[ii].split(sep="_")[3].split(sep='.')[0] 
+            if subject == sub2 and config == config2 and trial == trial2: 
+                counter = 1 
+        if counter == 0: 
+            parentList.append(entry) 
+            
+        
+
+for entry in parentList[1:3]:
+    print(entry )  
+    
+    # entry = parentList[6]
+    if "FootDorsum" in entry:  
+        fd_found = 1
+
+        subject = entry.split(sep="_")[0] 
+        config = entry.split(sep="_")[1]
+        trial = entry.split(sep="_")[3].split(sep='.')[0]  
+        tmpDat = readXSENSORFile(entry,fPath) 
+        plantDorDat = createTSmat(entry,fPath,tmpDat) 
+        
+        fb_found = 0 
+            
+        # For loop to find corresponding file in other list
+        for ii in range(len(entries_FrontBack)):
+            sub2 = entries_FrontBack[ii].split(sep="_")[0]
+            config2 = entries_FrontBack[ii].split(sep="_")[1]
+            trial2 = entries_FrontBack[ii].split(sep="_")[3].split(sep='.')[0]
+            
+
+            if (subject == sub2) and (config == config2) and (trial == trial2): 
+                print(entries_FrontBack[ii]) 
+                print(' ')
+                fb_found = 1
+                frontBackDat = createAvg_FB_Mat(entries_FrontBack[ii])  
+                
+    elif "FrontBack" in entry: 
+        fb_found = 1  
+        fd_found = 0 
+        frontBackDat = createAvg_FB_Mat(entry)  
+        
+                
+   
+        
+    answer = True
+    if data_check == 1:
+        if len(plantDorDat.LplantarMat) != 0:
+            plotAvgStaticFDPressure(plantDorDat.LplantarMat,plantDorDat,frontBackDat, fPath)
+        if len(plantDorDat.RplantarMat) != 0:
+            plotAvgStaticFDPressure(plantDorDat.RplantarMat,plantDorDat,frontBackDat , fPath)
```

Comment:
I do see that you were playing around with mis-matched files (I think you temporarily removed Conrads Intact shin 1). If the loop is started with conrad (index 6) there is an error in the debugging plot because it is expecting the frontbackdat. Is there a way in debugging to accommodate if one of the files is missing?